### PR TITLE
Track balance changes per wallet

### PR DESCRIPTION
**Pull request by philipjames44**

### CODE REVIEW

File: subgraphs/cellars/schema.graphql [hunk 1]
```diff
@@ -23,6 +23,14 @@ type Wallet @entity {
   dayDatas: [WalletDayData!]! @derivedFrom(field: "wallet")
```
Comment (sandbochs) [inline subgraphs/cellars/schema.graphql:23]: Referencing  the deposits here would be great.

Comment (philipjames44) [reply subgraphs/cellars/schema.graphql:23]: Added in

File: subgraphs/cellars/schema.graphql [hunk 2]
```diff
@@ -23,6 +23,14 @@ type Wallet @entity {
...
 }
 
+type Deposit @entity {
+  id: ID! # tx address
+  amount: BigInt!
+  date: Int!
+  depositerAddress: String!
+  cellarAddress: String!
```
[the remark this answers is not in the document]
Comment (philipjames44) [reply subgraphs/cellars/schema.graphql:32]: Added + abstracted away deposits + withdraws behind BalanceChange

File: subgraphs/cellars/subgraph.yaml [hunk 3]
```diff
@@ -27,6 +27,7 @@ dataSources:
         - CellarDayData
```
Comment (philipjames44) [inline subgraphs/cellars/subgraph.yaml:27]: Just wanted to post an example query + res here:


```
{
	wallet (id:"0x65c8b1562033a324359c1eceb1f4837a642669e7") {
		balanceChanges {
		  id
      kind
      shares
      shareValue
      amount
      date
      amount
		}
  }
}
```


```
{
  "data": {
    "wallet": {
      "balanceChanges": [
        {
          "id": "0x1ae547c91a0bc62762244caa8a56a325eeb1a557d36c5d48eb64ab561de89c97-292",
          "kind": "WITHDRAW",
          "shares": "100000000000000",
          "shareValue": "100000000",
          "amount": "100000000000000",
          "date": 1689261419
        },
        {
          "id": "0x40b201b91bc99a28c4847928c54916ea8192389056b3bab92a704a0346d7058b-254",
          "kind": "DEPOSIT",
          "shares": "100000000000000",
          "shareValue": "100000000",
          "amount": "100000000000000",
          "date": 1689260723
        },
        {
          "id": "0x47e4280c413e7bf3ead311ef2821108b7e1576a12c6105ad5299972a6b965a96-157",
          "kind": "DEPOSIT",
          "shares": "200000000000000",
          "shareValue": "100000000",
          "amount": "200000000000000",
          "date": 1689097343
        },
        {
          "id": "0xb64fe98fb63980435c706f9ad2d43224d8f372990e7236ea03b32fdf20f663c3-513",
          "kind": "DEPOSIT",
          "shares": "61000000000000",
          "shareValue": "100000000",
          "amount": "61000000000000",
          "date": 1689074807
        },
        {
          "id": "0xeba1f65fea491444b0c1e54ba7b38cb92787265e2db0b5a9a7fbdc061718fd30-343",
          "kind": "WITHDRAW",
          "shares": "100000000000000",
          "shareValue": "100000000",
          "amount": "100000000000000",
          "date": 1689135947
        },
        {
          "id": "0xefc92ab33bf9cd0cbdea92edb059c1b15a8355f46f7884925493f74f8e67fd44-137",
          "kind": "DEPOSIT",
          "shares": "61000000000000",
          "shareValue": "100000000",
          "amount": "61000000000000",
          "date": 1688797739
        }
      ]
    }
  }
}
```

**Pull request merged**